src/components/Authentication/Registration/__tests__/RegistrationPage.test.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+import { BrowserRouter } from 'react-router-dom';
+import { RegistrationPage, mapStateToProps } from '../RegistrationPage';
+
+
+const props = {
+  error: { email: ['email-error'], password: ['password-error'], username: ['username-error'] },
+  successMessage: 'success',
+  loading: null,
+  password: 'password',
+  repeat_password: 'password',
+  onSubmit: jest.fn(),
+};
+
+describe('Registration Page __tests__', () => {
+  it('Should return an error message', () => {
+    // arrange
+
+    const wrapper = mount(<BrowserRouter><RegistrationPage {...props} /></BrowserRouter>);
+    const submitButton = wrapper.find('#submit').last();
+    expect(wrapper.find('#submit').exists()).toBe(true);
+    submitButton.simulate('click');
+  });
+});
+
+describe('Registration Page class Methods', () => {
+  it('formatUserData Function', () => {
+    const setupProps = Object.assign({}, props, { actions: {}});
```

Comment:
Instead of using `Object.assign` add actions to the props object and pass the props to the component instead.

Reply:
Changed.